src/App.tsx

```diff
@@ -1,7 +1,40 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  key: string;
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    key: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({
+        key: event.key,
+      });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({
+        key: event.key,
+      });
+    });
```

Comment:
this won't remove listener, save your handler somewhere and pass it as a callback